analyticsdataserver/views.py

```diff
@@ -71,36 +71,55 @@ class HealthView(APIView):
 
    """
     permission_classes = (permissions.AllowAny,)
-
-    def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
-        OK = 'OK'
-        UNAVAILABLE = 'UNAVAILABLE'
-
-        overall_status = UNAVAILABLE
-        db_conn_status = UNAVAILABLE
-
+    STATUS_OK = 'OK'
+    STATUS_UNAVAILABLE = 'UNAVAILABLE'
+
+    def _get_connection_status(self, connection_name):
+        """
+        With the passed in database name, try to make connection to the database.
+        If database is able to connect successfully, return 'OK' status.
+        Otherwise, return 'UNAVAILABLE' status
+        """
+        db_status = self.STATUS_UNAVAILABLE
         try:
-            connection_name = getattr(settings, 'ANALYTICS_DATABASE', 'default')
             cursor = connections[connection_name].cursor()
             try:
                 cursor.execute("SELECT 1")
                 cursor.fetchone()
-
-                overall_status = OK
-                db_conn_status = OK
+                db_status = self.STATUS_OK
             finally:
                 cursor.close()
         except Exception:  # pylint: disable=broad-except
             pass
+        return db_status
+
+    def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        overall_status = self.STATUS_UNAVAILABLE
+
+        # First try the default database.
+        # The default database hosts user and auth info. Without it, no API would function
+
+        default_db_status = self._get_connection_status('default')
+
+        analytics_db_name = getattr(settings, 'ANALYTICS_DATABASE_V1')
+        if not analytics_db_name:
+            # If the v1 db is not defined, we use the default analytics db name setting instead
+            analytics_db_name = getattr(settings, 'ANALYTICS_DATABASE')
```

Comment:
while we are transitioning we need both to be defined to be healthy

and there are enterprise tables in the old analytics db that we aren't planning to move. So I think this should be if v1 check it, and always check v0